jobs/furnishings/src/furnishings/stage_processors/stage_one.py

```diff
@@ -0,0 +1,187 @@
+
+    async def process(self, batch_processing: BatchProcessing):
+        """Process batch_processing entry."""
+        furnishings = Furnishing.find_by(
+                batch_id=batch_processing.batch_id,
+                business_id=batch_processing.business_id
+                )
+        if not furnishings:
+            await self._send_first_round_notification(batch_processing)
+        else:
+            # send paper letter if business is still not in good standing after 5 days of email letter sent out
+            pass
+
+    async def _send_first_round_notification(self, batch_processing: BatchProcessing):
+        """Process first round of notification(email/letter)."""
+        # send email/letter notification for the first time
+        email = self._get_email_address_from_auth(batch_processing.business_identifier)
+        if email:
+            # send email letter
+            _, eligible_details = InvoluntaryDissolutionService.check_business_eligibility(
+                batch_processing.business_identifier, False
+                )
+            if eligible_details:
+                new_furnishing = self._create_new_furnishing(
+                    batch_processing,
+                    eligible_details,
+                    Furnishing.FurnishingType.EMAIL,
+                    email
+                    )
+                # notify emailer
+                await self._send_email(new_furnishing)
+        else:
+            # send paper letter if business doesn't have email address
+            pass
+
+    def _create_new_furnishing(
+            self,
+            batch_processing: BatchProcessing,
+            eligible_details: InvoluntaryDissolutionService.EligibilityDetails,
+            furnishing_type: Furnishing.FurnishingType,
+            email: str = None
+            ) -> Furnishing:
+        """Create new furnishing entry."""
+        business = batch_processing.business
+        if business.legal_type == Business.LegalTypes.EXTRA_PRO_A.value:
+            furnishing_name = (
+                Furnishing.FurnishingName.DISSOLUTION_COMMENCEMENT_NO_TR_XPRO
+                if eligible_details.transition_overdue
+                else Furnishing.FurnishingName.DISSOLUTION_COMMENCEMENT_NO_AR_XPRO
+            )
+        else:
+            furnishing_name = (
+                Furnishing.FurnishingName.DISSOLUTION_COMMENCEMENT_NO_TR
+                if eligible_details.transition_overdue
+                else Furnishing.FurnishingName.DISSOLUTION_COMMENCEMENT_NO_AR
+            )
+
+        grouping_identifier = self._get_grouping_identifier(furnishing_type)
+
+        new_furnishing = Furnishing(
+            furnishing_type=furnishing_type,
+            furnishing_name=furnishing_name,
+            batch_id=batch_processing.batch_id,
+            business_id=batch_processing.business_id,
+            business_identifier=batch_processing.business_identifier,
+            processed_date=datetime.utcnow(),
```

Comment:
I was thinking to make this `null` and the emailer can populate it when it processes it - whether it fails or succeeds in calling the notify api